tests/cocotb/test_basic_streamer.py

```diff
@@ -0,0 +1,265 @@
+# ---------------------------------
+# Copyright 2024 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+#
+# Description:
+# This tests the basic configuration of the streamer
+# found in ./util/cfg/streamer_cfg.hjson
+#
+# Sequence of tests:
+# 1. First test read and write to CSR registers
+# 2. Check if the output addresses of TCDM
+#    are correct and valid
+# ---------------------------------
+
+import cocotb
+from cocotb.triggers import RisingEdge
+from cocotb.clock import Clock
+from cocotb_test.simulator import run
+import pytest
+import snax_util
+import os
+import subprocess
+
+# Configurable design time parameters
+NARROW_DATA_WIDTH = 64
+TCDM_DEPTH = 256
+TCDM_REQ_PORTS = 12
```

Comment:
Be careful! This should match with the Streamer `tcmd_req` ports number

Reply:
I know! :'( I have no idea how to make this automatic... Not unless I read the files configuration file but that's just too automated. This is meant for the default one for now haha